clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesNamedAccountCredentials.groovy

```diff
@@ -0,0 +1,200 @@
+    AccountCredentialsRepository accountCredentialsRepository
+
+    Builder name(String name) {
+      this.name = name
+      return this
+    }
+
+    Builder environment(String environment) {
+      this.environment = environment
+      return this
+    }
+
+    Builder accountType(String accountType) {
+      this.accountType = accountType
+      return this
+    }
+
+    Builder context(String context) {
+      this.context = context
+      return this
+    }
+
+    Builder cluster(String cluster) {
+      this.cluster = cluster
+      return this
+    }
+
+    Builder user(String user) {
+      this.user = user
+      return this
+    }
+
+    Builder userAgent(String userAgent) {
+      this.userAgent = userAgent
+      return this
+    }
+
+    Builder kubeconfigFile(String kubeconfigFile) {
+      this.kubeconfigFile = kubeconfigFile
+      return this
+    }
+
+    Builder requiredGroupMembership(List<String> requiredGroupMembership) {
+      this.requiredGroupMembership = requiredGroupMembership
+      return this
+    }
+
+    Builder dockerRegistries(List<LinkedDockerRegistryConfiguration> dockerRegistries) {
+      this.dockerRegistries = dockerRegistries
+      return this
+    }
+
+    Builder namespaces(List<String> namespaces) {
+      this.namespaces = namespaces
+      return this
+    }
+
+    Builder credentials(KubernetesCredentials credentials) {
+      this.credentials = credentials
+      return this
+    }
+
+    Builder accountCredentialsRepository(AccountCredentialsRepository accountCredentialsRepository) {
+      this.accountCredentialsRepository = accountCredentialsRepository
+      return this
+    }
+
+    private KubernetesCredentials buildCredentials() {
+      Config config = KubernetesConfigParser.parse(kubeconfigFile, context, cluster, user, namespaces)
+      config.setUserAgent(userAgent)
+      if (namespaces == null || namespaces.isEmpty()) {
+        namespaces = Collections.singletonList(config.getNamespace())
+      }
+
+      for (LinkedDockerRegistryConfiguration registry : dockerRegistries) {
+        if (registry.getNamespaces() == null || registry.getNamespaces().isEmpty()) {
+          registry.setNamespaces(namespaces)
+        }
+      }
+
+      return new KubernetesCredentials(
+          new KubernetesApiAdaptor(name, config),
+          namespaces,
+          dockerRegistries,
+          accountCredentialsRepository,
+      )
+    }
+
+    KubernetesNamedAccountCredentials build() {
+      if (name == null || name.isEmpty()) {
+        throw new IllegalArgumentException("Account name for Kubernetes provider missing.")
+      }
+      if (dockerRegistries == null || dockerRegistries.size() == 0) {
+        throw new IllegalArgumentException("Docker registries for Kubernetes account " + name + " missing.")
+      }
+      kubeconfigFile = kubeconfigFile != null && kubeconfigFile.length() > 0 ?
+          kubeconfigFile : System.getProperty("user.home") + "/.kube/config"
```

Comment:
Style: `System.getProperty...` should go on the next line (basically anything after the ternary colon)